crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -192,6 +197,58 @@ async fn check_gas(
     }
 }
 
+/// Run the bytecode verifier with a timeout of `timeout_us`
+/// This function only fails if the verification does not complete within the timeout
+/// If the actual verification succeeds or fails within the timeout, the function still succeeds
+#[instrument(level = "trace", skip_all)]
+async fn run_timed_bytecode_verifier(
+    transaction: &TransactionData,
+    timeout_us: u64,
```

Comment:
nit: Can you use a `Duration` here to avoid having to indicate the unit in the variable name?